### PR TITLE
DNA: Deprecate tracks/client lib

### PR DESCRIPTION
**Pull request by lezama**
Deprecates tracks/client lib moving it to the compat package
Adds deprecation tests for libs and global defined functions

#### Testing instructions:
* run the tests
* try to use [woocommerce-services](https://github.com/Automattic/woocommerce-services/blob/5fb39adb9814bef92ea80374fe989207ca961e47/classes/class-wc-connect-tracks.php#L69) 
* add this snippet to your test site
```
add_action( 'plugins_loaded', function() {
    jetpack_tracks_get_identity( 1 );
}, 99 );
```
* make sure you get a deprecation notice

### CODE REVIEW

File: packages/compat/lib/tracks/client.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @deprecated 7.5.0 use Automattic\Jetpack\Tracking->tracks_record_event_raw instead
+ */
+function jetpack_tracks_record_event_raw( $event ) {
```
Comment (dereksmart) [inline packages/compat/lib/tracks/client.php]: should we `_deprecate_function()` these as well?

Comment (lezama) [reply packages/compat/lib/tracks/client.php]: yesss

Comment (lezama) [reply packages/compat/lib/tracks/client.php]: https://github.com/Automattic/jetpack/pull/12907/commits/2907733029a3da15930bc5dab1ae9fa40a463624

**Comment by gravityrail** [comment]
❤️!

**Comment by gititon** [comment]
I'm getting 
```PHP Fatal error:  Uncaught Error: Call to undefined method Automattic\Jetpack\Tracking::jetpack_tracks_get_identity() in /var/www/html/wp-content/plugins/jetpack/packages/compat/lib/tracks/client.php:16
Stack trace:
#0 /var/www/html/wp-content/plugins/woocommerce/includes/tracks/class-wc-tracks-client.php(146): jetpack_tracks_get_identity(1)
#1 /var/www/html/wp-content/plugins/woocommerce/includes/tracks/class-wc-tracks.php(96): WC_Tracks_Client::get_identity(1)
#2 /var/www/html/wp-content/plugins/woocommerce/includes/tracks/events/class-wc-orders-tracking.php(34): WC_Tracks::record_event('orders_view', Array)
#3 /var/www/html/wp-includes/class-wp-hook.php(286): WC_Orders_Tracking->track_orders_view('')
#4 /var/www/html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters('', Array)
#5 /var/www/html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#6 /var/www/html/wp-admin/admin.php(365): do_action('load-edit.php')
#7 /var/www/html/wp-admin/edit.php(10): require_once('/var/www/html/w...')
#8 {main}
  thrown in /var/www/html/wp-content/plugins/jetpack/packages/compat/lib/tracks/client.php on line 16
```

**Comment by gititon** [comment]
Tested well with the update.

**Review by gititon — APPROVED** [review]

**Comment by gravityrail** [comment]
I have not been able to trigger any issues with WooCommerce services as described on a Jurassic site.

**Comment by zinigor** [comment]
I haven't been able to reproduce the original problem either, sorry. Can you please paste a stack trace here so we know how to get to the point where it fails. The point in the code where you have linked to should not fatal because of a function existence check.

**Review by zinigor — APPROVED** [review]
Tested, works well after triggering the fatal manually using:
```
add_action( 'plugins_loaded', function() {
    jetpack_tracks_get_identity( 1 );
}, 99 );
```
Thanks Derek and Miguel!

**Comment by gravityrail** [comment]
Cherry picked to branch-7.5 in 58010c6f8f3538dc838985b17a4bd76cd2d032f4

**Comment by kraftbj** [comment]
r205835-wpcom

**Pull request merged**